qiime/sdk/artifact.py

```diff
@@ -185,8 +202,16 @@ def _parse_uuid(self, string):
         return uuid.UUID(hex=string)
 
     # TODO implement provenance parsing
-    def _parse_provenance(self, string):
-        return string
+    def _parse_provenance(self, provenance):
+        if isinstance(provenance, str):
```

Comment:
`provenance` can be expressed as `json`, right? if so, would it make sense for this branch to attempt a decode?

(i realize this is merged, asking in the event this is worth opening an issue to discuss)